scripts/tools/eslint-plugin-prettier-internal-rules/prefer-ast-path-getters.js

```diff
@@ -0,0 +1,86 @@
+"use strict";
+
+const selector = [
+  "CallExpression[arguments.length<2]",
+  "> MemberExpression",
+].join("");
+
+const messageId = "prefer-ast-path-getters";
+
+const getNodeFunctionNames = new Set(["getValue", "getNode"]);
+const getParentNodeFunctionName = "getParentNode";
+
+function isPathMemberExpression(node) {
+  return node.object?.name?.toLowerCase().endsWith("path");
+}
+
+module.exports = {
+  meta: {
+    type: "suggestion",
+    docs: {
+      url: "https://github.com/prettier/prettier/blob/main/scripts/tools/eslint-plugin-prettier-internal-rules/prefer-ast-path-getters.js",
+    },
+    fixable: "code",
+    messages: {
+      [messageId]: "Prefer {{ getter }} instead of {{ method }}.",
+    },
+  },
+  create(context) {
+    const report = (node, argumentsLength, getterName) => {
+      const getterMethodMap = new Map([
+        ["node", "`path.getValue()` or `path.getNode()`"],
+        ["parent", "`path.getParentNode()` or `path.getParentNode(0)`"],
+        ["grandparent", "`path.getParentNode(1)`"],
+      ]);
+      context.report({
+        node,
+        messageId,
+        data: {
+          getter: `\`path.${getterName}\``,
+          method: getterMethodMap.get(getterName),
+        },
+        fix: (fixer) => [
+          fixer.replaceTextRange(
+            [
+              node.property.range[0],
+              // remove `()` for CallExpression
+              node.property.range[1] + 2 + argumentsLength,
```

Comment:
```suggestion
              // remove `()` for CallExpression
              node.range[1],
```

We don't need make it 100% safe for edge cases, but this change will make the fix much safer,  user may add comments to explain the argument or code is not formatted.

```js
path.getParentNode(
) // <- not formatted.
```

```js
path.getParentNode(/* comment */ 1 )
```